### PR TITLE
Update public-metrics-reference.adoc

### PR DESCRIPTION
**Pull request by Feediver1**
## Description

Resolves https://redpandadata.atlassian.net/browse/DOC-750
Review deadline: **Nov 21**

## Page previews

[Preview](https://deploy-preview-870--redpanda-docs-preview.netlify.app/current/reference/public-metrics-reference/#redpanda_memory_available_memory)

## Checks

- [ ] New feature
- [ ] Content gap
- [x] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/reference/pages/public-metrics-reference.adoc [hunk 1]
```diff
@@ -208,7 +208,7 @@ The low watermark for available memory at process start.
 
 === redpanda_memory_free_memory
 
-Available memory in bytes.
+Total memory potentially available (free plus reclaimable memory) to a CPU shard (core), in bytes.
```
Comment (Feediver1) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
Total memory potentially available (includes both free and reclaimable memory) to a CPU shard (core), in bytes.
```

File: modules/reference/pages/public-metrics-reference.adoc [hunk 2]
```diff
@@ -208,7 +208,7 @@ The low watermark for available memory at process start.
 
 === redpanda_memory_free_memory
 
-Available memory in bytes.
+Total memory potentially available (includes both free and reclaimable memory) to a CPU shard (core), in bytes.
```
Comment (JFlath) [inline modules/reference/pages/public-metrics-reference.adoc]: I don't think this is correct.

`redpanda_memory_free_memory` = Unallocated memory

`redpanda_memory_available_memory` = Unallocated + reclaimable

The problem with the original wording for this line is that it uses "Available", which is confusing when we use that in the other metric name to indicate something different.

**Review by JFlath — CHANGES_REQUESTED** [review]

File: modules/reference/pages/public-metrics-reference.adoc (hunk 2 above)
Comment (Feediver1) [inline modules/reference/pages/public-metrics-reference.adoc]:
```suggestion
Total amount of unallocated memory for a CPU shard (core), in bytes.
```

Comment (Feediver1) [reply modules/reference/pages/public-metrics-reference.adoc]: @JFlath Better?

Comment (JFlath) [reply modules/reference/pages/public-metrics-reference.adoc]: I think we should pull the word `available` out. It has a specific meaning to us with memory metrics, so using it generically here is a source of confusion. Maybe just:

> Total amount of unallocated memory for a CPU shard (core), in bytes.

**Review by kbatuigas — APPROVED** [review]

**Review by JFlath — APPROVED** [review]

**Pull request merged**